src/porepy/numerics/fv/mpfa.py

```diff
@@ -691,24 +674,42 @@ def _local_discr(self, g, k, bnd, eta=None, inverter="numba", apertures=None):
             (np.ones_like(bound_faces), (bound_faces, bound_cells)),
             shape=(g.num_faces, g.num_cells),
         )
-        cell_contrib = remove_not_neumann * cell_contrib
+        cell_contrib = remove_not_neumann * hf2f.T * cell_contrib
 
         bound_pressure_cell = dp + cell_contrib
+        # Contribution to face pressure from sub-cell gradients, calculated as
+        # gradient times distance. Then further map to faces, and divide by number
+        # of contributions per face
 
         sgn_arr = np.zeros(g.num_faces)
         sgn_arr[bound_faces] = g.cell_faces[bound_faces].sum(axis=1).A.ravel()
-        sgn_mat = sps.diags(sgn_arr)
+        sgn_mat = sps.diags(hf2f.T * (sgn_arr))
 
-        bound_pressure_face_neu = (
-            sgn_mat * half_face_per_face * hf2f * pr_cont_grad_all * igrad * rhs_bound
-        )
+        bound_pressure_face_neu = sgn_mat * pr_cont_grad_all * igrad * rhs_bound
+
+        # sgn_mat = sps.diags(sgn_arr)
+        # bound_pressure_face_neu = (
+        #     sgn_mat * hf2f * pr_cont_grad_all * igrad * rhs_bound
+        # )
         # For Dirichlet faces, simply recover the boundary condition
         bound_pressure_face_dir = sps.diags(bnd.is_dir.astype(np.int))
 
         bound_pressure_face = (
             bound_pressure_face_dir + remove_not_neumann * bound_pressure_face_neu
         )
 
+        if not subface_rhs:
+            # In this case we set the value at a face, thus, we need to distribute the
```

Comment:
If returning boundary discretization in terms of subfaces is an option, how do you distinguish between different subfaces on a face? Say I want to assign a bc u = x + y, and use a certain eta for the continuity point, can I then assign the right bc value to the right subface?

Reply:
Short answer; yes.
Long answer: I think this is where all your complaints about documentation above leads to. This part should be properly documented, partly here, but it is also common with MPSA. The difficulties arises since this is more a hack to be able to set subface boundary conditions, and will only be used by people that are _very_ familiar with mps/fa. To make it something other than a hack I think we would need to introduce a proper concept of a subgrid (not just subcell_topology). The boundary condition will be defined at the continuity points according to subcell_topology.subfno_unique. So for your example:
```python
# Find the continuity points
eta = 1/3
cont_pnt = g.face_centers[:g.dim, s_t.fno_unique] + eta * (
    g.nodes[:g.dim, s_t.nno_unique] - g.face_centers[:g.dim, s_t.fno_unique]
    )
bc_val = cont_pnt[0] + count_pnt[1]
```